ReSharper.FSharp/src/FSharp.Psi/src/Impl/Cache2/Parts/EnumPart.cs

```diff
@@ -28,11 +28,23 @@ public IType GetUnderlyingType()
       return TypeFactory.CreateUnknownType(GetPsiModule());
     }
 
-    public IList<IField> Fields =>
-      ProcessSubDeclaration<IField, IEnumMemberDeclaration>(input =>
-        GetDeclaration() is IFSharpTypeDeclaration decl && decl.TypeRepresentation is IEnumRepresentation repr
-          ? repr.EnumMembers
-          : EmptyList<IEnumMemberDeclaration>.InstanceList);
+    public IEnumerable<IField> Fields
+    {
+      get
+      {
+        var declaration = GetDeclaration();
+        if (declaration != null &&
```

Comment:
This check is redundant because of the following one.
`GetDeclaration()` could also be inlined to the `if` check, since it's not used elsewhere.